### PR TITLE
fixed a few clippy issues

### PR DESCRIPTION
**Pull request by sphinxc0re**
- [ ] I have added a summary of my changes to the changelog

### CODE REVIEW

**Comment by sphinxc0re** [comment]
> Please leave `net` crate out of this.

I don't see, why I should. Could you elaborate?

**Comment by ddd-mtl** [comment]
@sphinxc0re Sure.
Since I'm holding coherence for `net` & `test_bin` I would like to have the freedom to code the way I see fit. If people which are not contributing to those crates starts re-arranging the code in their own style it will just hinder my ability to contribute and hold coherence.

**Comment by sphinxc0re** [comment]
@ddd-mtl what do you mean by "holding coherence"? Because if coherence means being non-compliant with the rest of the team, I'm pretty much against that. It seems to me like you are putting personal preference over generally acknowledged standards of code quality.

**Comment by lucksus** [comment]
Regarding coherence around net, what is the plan for growing that coherence to include the whole Holochain team again as it definitely is a crucial part of Holochain. The initial scaffolding approach of separating most of the networking code out into a separate process is that: a temporary scaffolding approach.

**Comment by ddd-mtl** [comment]
@sphinxc0re It means I'm making sure it is working reliabely and implementing new features.

Here are some facts that dissprove your last statement:
1. My code in this PR respects the 'team' style, since it is enforced by rustfmt in CI.
2. There has never been a team agreement on 'team' style. You are just enforcing rustfmt and clippy defaults.
3. I have accepted rustfmt defaults despite not being my preference. [Proof](https://github.com/holochain/holochain-rust/pull/211)
4. There has been a team agreement on clippy configuration during mtl hodlpodl, which has not been enforced ([Proof](https://trello.com/c/Pk9f9j7F/18-the-clippy-linter-enforces-requirements-that-not-all-buy-into)).
5. You are manually enforcing clippy lint rules that are contrary to that team agreement (proof: this PR)

**Comment by lucksus** [comment]
Just for the record, during the last in-person hodl-podl team meeting we (that is everybody who came, and I know you, @ddd-mtl, opted out) decided unanimously that we want to turn clippy on with its defaults and had pushed on actually doing so because of more pressing issues and because of this change actually being a bit of work after living without clippy since October.

File: net/src/lib.rs [hunk 1]
```diff
@@ -1,6 +1,7 @@
 #![feature(fnbox)]
 #![feature(try_from)]
 #![feature(vec_remove_item)]
+#![allow(clippy::all)] // As per the request of the networking subteam
```
Comment (ddd-mtl) [inline net/src/lib.rs]: 👏

**Pull request merged**